macos/FluentUI/Button/Button.swift

```diff
@@ -5,158 +5,254 @@
 
 import AppKit
 
+// MARK: - Button Formatting Parameters
+
+/// Indicates the size of the button
+@objc(MSFButtonSize)
+public enum ButtonSize: Int, CaseIterable {
+	case large
+	case small
+}
+
 /// Indicates what style our button is drawn as
 @objc(MSFButtonStyle)
 public enum ButtonStyle: Int, CaseIterable {
-	case primaryFilled	// Solid fill color
-	case primaryOutline	// Clear fill color, solid outline
-	case borderless		// Clear fill color, clear outline
+	/// No defined colors; user to specify.
+	case none
+
+	/// Accent color fill, white text/image.
+	case primary
+
+	/// Light mode: white fill, black text/image, light gray outline.
+	/// Dark mode: dark gray fill, white text/image, subtle gray outline.
+	/// Pressed: accent color fill, white text/image, no outline (same as primary style).
+	case secondary
+
+	/// Light mode: light gray fill, black text/image.
+	/// Dark mode: dark gray fill, white text/image.
+	case acrylic
+
+	/// Accent color text/image, no fill or outline.
+	case borderless
+}
+
+/// Combination of all formatting parameters.
+public struct ButtonFormat {
+	var size: ButtonSize
+	var style: ButtonStyle
+	var accentColor: NSColor?
+
+	public init(
+		size: ButtonSize = .large,
+		style: ButtonStyle = .primary,
+		accentColor: NSColor? = Colors.primary
+	)
+	{
+		self.size = size
+		self.style = style
+		self.accentColor = accentColor
+	}
+}
```

Comment:
I might be wrong but I think we were moving away from structs to contain properties? @rdeassis or @anandrajeswaran might know more.

I previously said that for controls we want to wrap in React Native, we need to keep all the properties as a flat list instead of having a struct as a property that contained other properties. I was wrong, and both should work, and I just wanted to call that out.

Reply:
I don't know of any broad guidance on this, but I've generally found it easier to just use flat lists of elements since that tends to translate best for our objC consumers and allow selective updates - some property changes invalidate our intrinsicContentSize and some invalidate our display and some do both. This way we don't need to do both for all changes. Those are admittedly small benefits, but I can't think of any practical benefits of grouping together in a struct unless we are suggesting that all of them should logically only be used together

Reply:
@anandrajeswaran @Saadnajmi I cannot visualize the "flat list" structure you are describing.  Is there an example you could point me to?

Reply:
Instead of:

- Class
  - prop 1
  - prop 2
  - prop that is struct
    - prop 3
    - prop 4
    - prop 5

Do:

- Class
  - prop 1
  - prop 2
  - prop 3
  - prop 4
  - prop 5

Reply:
Here's a PR where I did that for Shimmer. In hindsight, I didn't *need* to (now that I understand memo caches in React Native better), but at the time I thought it was necessary https://github.com/microsoft/fluentui-apple/pull/215

Reply:
OK, I understand.  But in this case, `format` is a synthesized property that just maps to and from its three components without actually storing a struct anywhere in the class.  The ButtonFormat struct only exists because I found it really convenient to have predefined sets of properties that I could reuse and apply to buttons as a unit.  Examples the test app:

`let largePrimary = ButtonFormat(size: .large, style: .primary, accentColor: communicationBlue)`

`formats.map({ TestButton(image: stopImage, format: $0) })`

`TestButton(title: "Back", image: leadingArrowImage, imagePosition: .imageLeading, format: largePrimary)`

`TestButton(title: "Nope", image: stopImage, imagePosition: .imageAbove, format: largePrimary)`

Because the components are only transmitted and not stored as a struct, we aren't subject to the perf impact Anand mentioned: each of the properties will be set independently and can decide on its own what needs to be invalidated.

Reply:
This is a great discussion and love the care and attention to detail going into API design! I definitely agree that a struct to coalesce coupled properties is a great idea and that certainly might be the right design here. The two things to think of are
1) Will many clients want to only update some of the values in the struct? If most clients will only want to adjust one or some of the properties but then need to make a struct with all the properties to do so we may be adding unnecessary complexity and verbosity to the clients
2) What does the objC interface look like? The subtleties of structs vs. classes don't translate to objC well, so there's a chance doing this could put objC consumers in a bind.

If neither of the above are relevant (and I'm correctly understanding @Saadnajmi 's point that there is no impact of this decision on JS consumers in fluentui-react-native), then I really like the idea of a struct to tie together commonly coupled configs. Otherwise we may be adding more complexity than we're providing benefit.

Reply:
1. Yes, certainly clients may want to update only some values of the struct.  But the client never needs to make a struct in order to update one of the values.

2. The ObjC properties are the individual components of the struct:
@objc public var accentColor: NSColor = Colors.primary
@objc open var style: ButtonStyle = .primary
@objc public var size: ButtonSize = .large

And the ObjC initializers use those properties:
@objc public convenience init(title: String, size: ButtonSize = .large, style: ButtonStyle = .primary, accentColor: NSColor = Colors.primary)
@objc public convenience init(image: NSImage, size: ButtonSize = .large, style: ButtonStyle = .primary, accentColor: NSColor = Colors.primary)
@objc public convenience init(title: String, image: NSImage, imagePosition: NSControl.ImagePosition = .imageLeading, size: ButtonSize = .large, style: ButtonStyle = .primary, accentColor: NSColor = Colors.primary)
@objc public convenience init()